expes/hypergradient/plot_hypergradient_svm.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.sparse import csr_matrix
+from sklearn.model_selection import StratifiedShuffleSplit
+
+from lightning.classification import LinearSVC
+from libsvmdata import fetch_libsvm
+from scipy.sparse.linalg import norm
+
+from sparse_ho.models import SVM
+from sparse_ho.criterion import HeldOutSmoothedHinge
+from sparse_ho import ImplicitForward, Implicit
+from sparse_ho import Forward
+from sparse_ho.utils import Monitor
+
+maxits = [5, 10, 25, 50, 75, 100]
+# maxits = [5, 10, 25, 50, 75, 100, 500, 1000]
+methods = ["forward", "implicit_forward", "sota"]
+
+dict_label = {}
+dict_label["forward"] = "forward"
+dict_label["implicit_forward"] = "Implicit"
+dict_label["sota"] = "Implicit + sota"
+
+
+logC = np.log(0.15)
+
+tol = 1e-32
+
+dataset_name = "real-sim"
+X, y = fetch_libsvm(dataset_name)
+X = X[:, :2000]
+X = csr_matrix(X)  # very important for SVM
+my_bool = norm(X, axis=1) != 0
+X = X[my_bool, :]
+y = y[my_bool]
+
+sss1 = StratifiedShuffleSplit(n_splits=2, test_size=0.3333, random_state=0)
+idx_train, idx_val = sss1.split(X, y)
+idx_train = idx_train[0]
+idx_val = idx_val[0]
+
+
+true_monitor = Monitor()
+clf = LinearSVC(
+        C=np.exp(logC), tol=1e-32, max_iter=10_000, loss='hinge',
+        permute=False)
+criterion = HeldOutSmoothedHinge(idx_train, idx_val)
+algo = Implicit(criterion)
+model = SVM(estimator=clf, max_iter=10_000)
+true_val, true_grad = criterion.get_val_grad(
+        model, X, y, logC, algo.get_beta_jac_v, tol=1e-14,
+        monitor=true_monitor)
+
+dict_res = {}
+for max_iter in maxits:
+    for method in methods:
+        print("Dataset %s, maxit %i" % (method, max_iter))
+        for i in range(2):
+            monitor = Monitor()
+            model = SVM(max_iter=max_iter)
+            criterion = HeldOutSmoothedHinge(idx_train, idx_val)
+            if method == "sota":
+                clf = LinearSVC(
+                    C=np.exp(logC), loss='hinge', max_iter=max_iter, tol=1e-32,
+                    permute=False)
```

Comment:
No need to put fit_intercept=False ?

Reply:
I might have missed something, but a priori there is no fit intercept=True to pass to the `LinearSVC` from Lightning:
https://contrib.scikit-learn.org/lightning/generated/lightning.classification.LinearSVC.html